weave/tests/test_dataset.py

```diff
@@ -0,0 +1,125 @@
+import pytest
+
+import weave
+
+
+def test_dataset_append_basic(client):
+    ds = weave.Dataset(rows=[{"a": 1, "b": 2}])
+    ds.append({"a": 3, "b": 4})
+    ref = weave.publish(ds)
+
+    ds2 = ref.get()
+    assert ds2 == [
+        {"a": 1, "b": 2},
+        {"a": 3, "b": 4},
+    ]
+    ds2.append({"a": 5, "b": 6})
+    ref2 = weave.publish(ds2)
+
+    ds3 = ref2.get()
+    assert ds3 == [
+        {"a": 1, "b": 2},
+        {"a": 3, "b": 4},
+        {"a": 5, "b": 6},
+    ]
+
+
+@pytest.mark.xfail(reason="Table API does not properly handle objects yet")
```

Comment:
TODO in a follow up PR.  This is related to Thomas's `Point(x,y)` dataset issue:
https://wandb.atlassian.net/browse/WB-20236